### PR TITLE
issue/7 - check broker is available before connecting to emitters

### PR DESCRIPTION
**Pull request by stormeye2000**

### CODE REVIEW

File: src/main/java/com/stormeye/producer/config/AppConfig.java [hunk 1]
```diff
@@ -33,15 +35,20 @@ public class AppConfig {
     private String clientId;
     final ServiceProperties properties;
 
+    final static int PRODUCER_BYTES = 33554432;
```
Comment (meywood) [inline src/main/java/com/stormeye/producer/config/AppConfig.java]: Should be private

File: src/main/java/com/stormeye/producer/config/BrokerState.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    private final AdminClient adminClient;
+
+    public BrokerState(final AdminClient adminClient) {
+        this.adminClient = adminClient;
+    }
+
+    public boolean isAvailable() {
+        try {
+            final ListTopicsResult topics = adminClient.listTopics();
+            topics.names().get();
```
Comment (meywood) [inline src/main/java/com/stormeye/producer/config/BrokerState.java]: What does this do ` topics.names().get()`?

**Review by meywood — CHANGES_REQUESTED** [review]

File: src/main/java/com/stormeye/producer/config/AppConfig.java (hunk 1 above)
Comment (stormeye2000) [reply src/main/java/com/stormeye/producer/config/AppConfig.java]: done

File: src/main/java/com/stormeye/producer/config/BrokerState.java (hunk 2 above)
Comment (stormeye2000) [reply src/main/java/com/stormeye/producer/config/BrokerState.java]: it queries the topics returned, i could use a local var but didn't think necessary

Comment (meywood) [reply src/main/java/com/stormeye/producer/config/BrokerState.java]: The shouldn't it do this `return !topics.names().get().isEmpty();`?

Comment (stormeye2000) [reply src/main/java/com/stormeye/producer/config/BrokerState.java]: done, merge it if you can and it'll deploy

**Pull request closed without merging**